cpu/mips_pic32_common/Makefile.include

```diff
@@ -1,7 +1,31 @@
+include $(RIOTCPU)/mips32r2_common/Makefile.include
+
 export INCLUDES += -I$(RIOTCPU)/mips_pic32_common/include
 
 USEMODULE += mips_pic32_common
 USEMODULE += mips_pic32_common_periph
 
 USEMODULE += periph_common
 USEMODULE += periph_hwrng
```

Comment:
Now that you clean up the Makefiles what is this line supposed to do? If this is a dependency it should go into a Makefile.dep. If it means that the hardware has an hwrng on board, it should be `FEATURES_PROVIDED += periph_hwrng` right?

Reply:
I have no Idea how RIOT's make system is supposed to actually work, Its not really documented very well, I mean what is the difference between 'USEMODULE' and 'FEATURES_PROVIDED' and does it matter which you use ?

In answer to your question (on another thread) yes PIC32 devices support HWRNG and a driver is implemented.

Reply:
The FEATURES_PROVIDED list the features that a particular board can offer, while the USEMODULE explicitly tells the build system to build that support. FEATURES_PROVIDED is set by the board and CPU makefiles, while the application and its dependencies decide whether it should be compiled in or not. This is done in Makefile.dep for implicit dependencies and additional application specific dependencies are listed in the main Makefile for the application.

Reply:
Additionally, FEATURES_REQUIRED is used by an application to state that it needs a particular set of hardware/driver features in order to run properly. See the Makefiles in tests/periph_xxx for an example of this.

Reply:
The features in `Makefile.features` for all cpu of the mips family have been added by https://github.com/RIOT-OS/RIOT/commit/28a2028e1e8f099853f6b238b526345c2746a6b0
And may not be complete.

I think in this case, it should just also be added to the `mips_pic32_common/Makefile.features`:
```
FEATURES_PROVIDED += periph_hwrng
```
Is that right @gebart ?

But this could be a different PR, "update features for mips cpus".